utils/constants.go

```diff
@@ -47,6 +48,7 @@ var (
 					Interval:     kong.Int(0),
 					HTTPStatuses: []int{429, 404, 500, 501, 502, 503, 504, 505},
 				},
+				HTTPSVerifyCertificate: kong.Bool(true),
```

Comment:
(Blocker comment)
Please verify in test that the user can set this value to false in the state file and that it doesn't get set to `true` i.e. this value can be set to `false` in the file and that gets reflected in Kong's database.

Reply:
when set HTTPSVerifyCertificate true in constants.go,  https://github.com/Kong/deck/blob/19205e5408661377f079960f198128743d716b36/utils/defaulter_test.go#L355 test can't pass, when set HTTPSVerifyCertificate false in arg and want, mergo.Merge will always set HTTPSVerifyCertificate to true according https://github.com/imdario/mergo/issues/24

Reply:
@hbagdi  I don't know "set this to false in the state file" exact  meaning and kongingress crd don't have HTTPSVerifyCertificateso param, so I code in getUpstreams func.
 https://github.com/Kong/kubernetes-ingress-controller/blob/2caa5240c2f10ade2781b0edf37250e3f38ab927/internal/ingress/controller/parser/parser.go#L1291
like 
![image](https://user-images.githubusercontent.com/1832550/116219700-75dbf780-a77e-11eb-8a54-5d4a216418c7.png)
getUpstream called in Build func, Build func creates a Kong configuration from Ingress and Custom resources defined in Kuberentes, so I think this change can simulate user change the  HTTPSVerifyCertificate value in state file?
then I print upstream data before and after file.Get func in https://github.com/Kong/kubernetes-ingress-controller/blob/2caa5240c2f10ade2781b0edf37250e3f38ab927/internal/ingress/controller/kong.go#L265
 
![image](https://user-images.githubusercontent.com/1832550/116220525-572a3080-a77f-11eb-9fae-c9150b76b7f5.png)

HTTPSVerifyCertificate really set to false. and I check kong config by admin api 
![image](https://user-images.githubusercontent.com/1832550/116221161-f2230a80-a77f-11eb-8ec2-fdc96883b511.png)

so this value can be set to false and reflected in Kong's database, right?

Reply:
@hbagdi  the next step is check and get this "Healthchecks.Active.HTTPSVerifyCertificate can be set to false"  test pass?

Reply:
> (Blocker comment)
> Please verify in test that the user can set this value to false in the state file and that it doesn't get set to `true` i.e. this value can be set to `false` in the file and that gets reflected in Kong's database.

I have test deck v1.1.0 which kic 0.9.1 used, I add default value logic to constants.go, I don't understand "set this value to false in the state file" exact meaning, so I add code in getUpstreams func
https://github.com/Kong/kubernetes-ingress-controller/blob/2caa5240c2f10ade2781b0edf37250e3f38ab927/internal/ingress/controller/parser/parser.go#L1291
![image](https://user-images.githubusercontent.com/1832550/116238486-f822e680-a793-11eb-98d4-bf061fcb1c39.png)


this func called by Build func in https://github.com/Kong/kubernetes-ingress-controller/blob/0.9.x/internal/ingress/controller/parser/parser.go, Build func creates a Kong configuration from Ingress and Custom resources defined in Kuberentes, So I think this operator simulate user set value in the state file.

when I print data before and after file.Get func in https://github.com/Kong/kubernetes-ingress-controller/blob/2caa5240c2f10ade2781b0edf37250e3f38ab927/internal/ingress/controller/kong.go#L265
 
the same upstream is change, yes, current is false and after file.Get func is become true
![image](https://user-images.githubusercontent.com/1832550/116239072-ccecc700-a794-11eb-9ba1-be99d53270bc.png)

so I think when no default value logic, the HTTPSVerifyCertificate value in state file will remain, file.Get has no bug. but when add default value logic, file.Get always overridden  HTTPSVerifyCertificate value. that's why "Healthchecks.Active.HTTPSVerifyCertificate can be set to false" test can't pass